apps/fortuna/src/api.rs

```diff
@@ -43,7 +43,7 @@ pub struct ApiMetrics {
 
 #[derive(Clone)]
 pub struct ApiState {
-    pub chains: Arc<HashMap<ChainId, BlockchainState>>,
+    pub chains: Arc<RwLock<HashMap<ChainId, BlockchainState>>>,
```

Comment:
you can use Dashmap for a concurrent hashmap https://docs.rs/dashmap/latest/dashmap/struct.DashMap.html

Reply:
Also I think this should be a `HashMap<ChainId, Option<BlockchainState>>` so "chain is uninitialized" isn't conflated with "unknown chain id".  You could make BlockchainState an enum or something as well to handle this.